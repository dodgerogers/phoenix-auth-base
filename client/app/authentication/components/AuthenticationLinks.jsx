import React from 'react'
import PropTypes from 'prop-types';
import mirrorCreator from 'mirror-creator';
import { ModalIds, ModalTrigger } from '../../common/modals';
import styled from 'styled-components';

const AuthLink = styled.div`
  margin-top: 10px;
`;

export const links = mirrorCreator([
  'LOGIN',
  'REGISTRATION',
  'CONFIRMATION',
  'RESEND_CONFIRMATION',
]);

const AuthenticationLinks = (props) => {
  const Wrapper = props => (<a href="#" {...props} />);
  const authLinksMap = {
    [links.LOGIN]: () => (
      <ModalTrigger
        text="Already registered? Login in"
        id={ModalIds.LOGIN_MODAL}
        wrapper={Wrapper}
      />
    ),
    [links.REGISTRATION]: () => (
      <ModalTrigger
        text="Create an account"
        id={ModalIds.REGISTRATION_MODAL}
        wrapper={Wrapper}
      />
    ),
    [links.CONFIRMATION]: () => (
      <ModalTrigger
        text="I have a confirmation code"
        id={ModalIds.CONFIRMATION_MODAL}
        wrapper={Wrapper}
      />
    ),
    [links.RESEND_CONFIRMATION]: () => (
      <ModalTrigger
        text="Didn't receive confirmation instructions?"
        id={ModalIds.RESEND_CONFIRMATION_MODAL}
        wrapper={Wrapper}
      />
    ),
  }

  const renderLinks = () => {
    const filterOutExcludedLink = link => props.exclude.indexOf(link) < 0;
    return Object.keys(authLinksMap).filter(filterOutExcludedLink).map(link => {
      return (
        <AuthLink key={link}>{authLinksMap[link]()}</AuthLink>
      );
    });
  }

  return (
    <div className="authentication-links">
      {renderLinks()}
    </div>
  );
}

AuthenticationLinks.propTypes = {
  exclude: PropTypes.array,
};

AuthenticationLinks.defaultProps = {
  exclude: [],
};

export default AuthenticationLinks;
