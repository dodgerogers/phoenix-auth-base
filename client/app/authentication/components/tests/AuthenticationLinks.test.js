import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AuthenticationLinks from '../AuthenticationLinks';


describe('AuthenticationLinks', () => {
  const fullProps = {
    close: jest.fn(),
  };

  const pureComponent = (props) => shallow(
    <AuthenticationLinks />
  );

  it('matches snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
