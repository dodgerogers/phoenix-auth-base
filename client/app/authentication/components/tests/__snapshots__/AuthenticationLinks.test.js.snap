// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationLinks matches snapshot 1`] = `
<div
  className="authentication-links"
>
  <styled.div>
    <Connect(ModalTrigger)
      id="CONFIRMATION_MODAL"
      text="I have a confirmation code"
      wrapper={[Function]}
    />
  </styled.div>
  <styled.div>
    <Connect(ModalTrigger)
      id="RESEND_CONFIRMATION_MODAL"
      text="Didn't receive confirm instructions?"
      wrapper={[Function]}
    />
  </styled.div>
</div>
`;
